pkg/roles/provisionmanage.go

```diff
@@ -559,6 +559,16 @@ func (r ProvisionManageService) RoleRemoveMembers(ctx context.Context, session a
 		return errors.Wrap(svcerr.ErrRemoveEntity, err)
 	}
 
+	if _, ok := r.builtInRoles[BuiltInRoleName(ro.Name)]; ok {
+		page, err := r.repo.RoleListMembers(ctx, ro.ID, 0, 0)
+		if err != nil {
+			return errors.Wrap(svcerr.ErrViewEntity, err)
+		}
+		if page.Total <= uint64(len(members)) {
+			return errors.Wrap(svcerr.ErrRemoveEntity, fmt.Errorf("built-in role '%s' must retain at least one member", ro.Name))
+		}
+	}
```

Comment:
This should be applicable only for `domains`
We can move this logic to domain  service `func (svc *service) RoleRemoveAllMembers(ctx context.Context, session authn.Session, entityID, roleID string) error`